lis/dataassim/algorithm/enkf/enkf_general.F90

```diff
@@ -184,7 +184,15 @@ subroutine enkf_analysis( gid, &
        ! compute right hand side rhs = Zpert - H*y^f for system equation,
        
        do i=1,N_obs
-          rhs(i) = Observations(i)%value + Obs_err(i,n_e) - Obs_pred(i,n_e)
+          if(Observations(i)%pert_type.eq.0) then  ! additive observation error
+            rhs(i) = Observations(i)%value + Obs_err(i,n_e) - Obs_pred(i,n_e)
+          elseif(Observations(i)%pert_type.eq.1) then ! multiplicative observation error 
+            if(Observations(i)%value.eq.0) then ! if the observation = 0, multiply by model estimate instead
+              rhs(i) = Observations(i)%value + (Obs_pred_bar(i) * Obs_err(i,n_e)) - Obs_pred(i,n_e)
```

Comment:
I get why something needs to be done for value = 0. 
Two questions:
- Why should the model estimate be used?
- For a value slightly above 0, one would switch to the else statement which results in an abrupt change of the error from one time step to the other. This could be avoided by using some low constant error value for very small observation values.

Reply:
I think this answers both questions: I wanted this fix to be widely applicable for different applications with different ranges of values. Before this change, if the we had a value = 0, the value was hardcoded to 0.1. This then has very different implications depending on if you're working with a variable with a small or large variance.

The same answer I think applies to the second question. if we assume a constant low error value for very small observation values this would look different for different variables. For example 0.1 may be a 'low value' for snow but not for a different variable. Similarly, 0.1 may be a low error value for snow but not for a different variable.

I know this does result in an abrupt change of the error from one time step to the other but I think it also makes it more widely applicable